### ISSUE TITLE
`auth:cancel` succeeds, but abandons applications.

### ISSUE THREAD

**Issue by slack** [deis/workflow#213]
If you have a set of owned applications and run `deis auth:cancel` the call succeeds, deleting the user and applications, but does not deprovision the RC/Pods.

auth:cancel:
```console
example-go [master]$ deis2 auth:cancel
Please log in again in order to cancel this account
username: jhansen
password:
Logged in as jhansen
cancel account jhansen at http://deis.chorizo.k8s.slack.io? (y/N): y
Account cancelled
example-go [master]$
```

Abandoned applications:
```console
example-go [master]$ k get namespaces
NAME              LABELS          STATUS    AGE
default           <none>          Active    9d
deis              heritage=deis   Active    6d
kube-system       <none>          Active    9d
owlish-rabbitry   <none>          Active    5d
youthy-turbojet   <none>          Active    5d
example-go [master]$ k --namespace=owlish-rabbitry get po
NAME                           READY     STATUS    RESTARTS   AGE
owlish-rabbitry-v2-web-2fyby   1/1       Running   0          5d
owlish-rabbitry-v2-web-5plh1   1/1       Running   0          5d
owlish-rabbitry-v2-web-6eup6   1/1       Running   0          5d
[ REDACTED ]
owlish-rabbitry-v2-web-xszgd   1/1       Running   0          5d
owlish-rabbitry-v2-web-yzb2d   1/1       Running   0          5d
owlish-rabbitry-v2-web-zg90h   1/1       Running   0          5d
```

We should either reject cancellation for users with owned applications or tear everything down.

**Comment by slack** [comment]
Also, if you re-create the app using the same name, all sorts of strange things happen with the left-over ReplicationController.

**Comment by bacongobbler** [comment]
that's a curious bug. Since it's all Django model logic the namespace as well as all RCs and pods should be removed (this was the intended behaviour for fleet as well). Does `deis apps:delete` still work as intended?

**Comment by slack** [comment]
Yep, `apps:destroy` works as expected:
```console
example-go [master]$ k --namespace=jagged-yachting get rc,po
CONTROLLER               CONTAINER(S)          IMAGE(S)                             SELECTOR                                                REPLICAS   AGE
jagged-yachting-v9-web   jagged-yachting-web   quay.io/deisci/slugrunner:v2-alpha   app=jagged-yachting,heritage=deis,type=web,version=v9   2          9m
NAME                           READY     STATUS    RESTARTS   AGE
jagged-yachting-v9-web-7z3ay   1/1       Running   0          9m
jagged-yachting-v9-web-htu50   1/1       Running   0          9m
example-go [master]$ deis2 apps:destroy
 !    WARNING: Potentially Destructive Action
 !    This command will destroy the application: jagged-yachting
 !    To proceed, type "jagged-yachting" or re-run this command with --confirm=jagged-yachting

> jagged-yachting
Destroying jagged-yachting...
done in 1s
Git remote deis removed
example-go [master]$ k --namespace=jagged-yachting get rc,po
CONTROLLER   CONTAINER(S)   IMAGE(S)   SELECTOR   REPLICAS   AGE
NAME      READY     STATUS    RESTARTS   AGE
example-go [master]$
```

**Comment by helgi** [comment]
> We should either reject cancellation for users with owned applications or tear everything down

Which behaviour do we want?

**Comment by slack** [comment]
Since account cancellation with active applications is broadly destructive, we should throw a `409 Conflict`.

**Comment by helgi** [comment]
Are we wanting to do any messaging like "transfer or delete the associated apps with this user" ?

**Comment by helgi** [comment]
https://github.com/deis/deis/issues/4909 and https://github.com/deis/deis/issues/4896 are related

**Comment by slack** [comment]
Yes, wherever we can present meaningful error messages :+1:

**Comment by helgi** [comment]
Fixed by #391
